org.civicrm.reqangular/src/common/services/file-upload.js

```diff
@@ -65,9 +65,28 @@ define([
             formData: [{
               entity_table: customSettings.entityTable || error('entityTable'),
               crm_attachment_token: customSettings.crmAttachmentToken || error('crmAttachmentToken')
+            }],
+            filters: [{
+              name: 'fileFormatFilter',
+              fn: function (item /*{File|FileLikeObject}*/ , options) {
+                var type = '|' + item.type.slice(item.type.lastIndexOf('/') + 1) + '|';
+
+                //checks for mime types and not file name extensions
+                //|txt|jpg|png|jpeg|bmp|gif|pdf|doc|docx|xls|xlsx|ppt|pptx|
+                return '|plain|png|jpeg|bmp|gif|pdf|msword|vnd.openxmlformats-officedocument.wordprocessingml.document|vnd.ms-excel|vnd.openxmlformats-officedocument.spreadsheetml.sheet|vnd.ms-powerpoint|vnd.openxmlformats-officedocument.presentationml.presentation|'.indexOf(type) !== -1;
```

Comment:
I don't think this is a scalable approach, you are basically imposing the list of allowed file types for any application that wants to use the file uploader service.

What instead should happen is that the application itself defines the list of file types that it can accept, and then pass the information to the file uploader